azurerm/resource_arm_databricks_workspace.go

```diff
@@ -0,0 +1,244 @@
+	return &schema.Resource{
+		Create: resourceArmDatabricksWorkspaceCreate,
+		Read:   resourceArmDatabricksWorkspaceRead,
+		Update: resourceArmDatabricksWorkspaceUpdate,
+		Delete: resourceArmDatabricksWorkspaceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"workspace_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+				Default:  "standard",
+				ValidateFunc: validation.StringInSlice([]string{
+					"standard",
+					"premium",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDatabricksWorkspaceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).databricksWorkspacesClient
+	ctx := meta.(*ArmClient).StopContext
+	log.Printf("[INFO] preparing arguments for Azure ARM Databricks Workspace creation.")
+
+	subscriptionID := meta.(*ArmClient).subscriptionId
+	name := d.Get("workspace_name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resGroup := d.Get("resource_group_name").(string)
+	skuName := d.Get("sku").(string)
+
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	managedResourceGroupID := fmt.Sprintf("/subscriptions/%s/resourceGroups/databricks-rg-%s", subscriptionID, resGroup)
+
+	properties := databricks.WorkspaceProperties{
+		ManagedResourceGroupID: &managedResourceGroupID,
+	}
+
+	workspace := databricks.Workspace{
+		Sku: &databricks.Sku{
+			Name: &skuName,
+		},
+		Location:            &location,
+		WorkspaceProperties: &properties,
+		Tags:                expandedTags,
+	}
+
+	future, err := client.CreateOrUpdate(ctx, workspace, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error Creating/Updating Databricks Workspace %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error Retrieving Databricks Workspace %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Databricks Workspace Instance %s (resource group %s) ID", name, resGroup)
+	}
+
+	log.Printf("[DEBUG] Waiting for Databricks Workspace (%s) to become available", name)
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Updating", "Creating"},
+		Target:     []string{"Succeeded"},
+		Refresh:    workspaceStateRefreshFunc(ctx, client, resGroup, name),
+		Timeout:    60 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Databricks Workspace (%s) to become available: %s", name, err)
+	}
```

Comment:
given there's a Future above which waits for the completion of this resource - do we need this separate poller?